Src/Couchbase.Linq/N1QlFunctions.Core.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Couchbase.Linq
+{
+    /// <summary>
+    /// Implements static helper methods for N1QL queries
+    /// </summary>
+    public static partial class N1QlFunctions
+    {
+        /// <summary>
+        /// Shortcut for creating an error for methods that are only supported in N1QL, not in .Net.
+        /// </summary>
+        private static Exception NotImplementedError()
+        {
+            return new NotImplementedException("This method may only be used for generating N1QL queries.");
```

Comment:
I am not sure I understand the message here...not implemented means, well, not implemented. How can the method this exception is thrown from have an implementation (assumed from "This method may only be used for generating N1QL queries.")?